packages/rsocket-core/src/RSocketMachine.js

```diff
@@ -943,3 +943,11 @@ function deserializeMetadataPushPayload<D, M>(
     metadata: serializers.metadata.deserialize(frame.metadata),
   };
 }
+
+export class RSocketError extends Error {
```

Comment:
This might be best as its own file `RSocketError.js`.

Reply:
Done 👍🏽